### ISSUE TITLE
Possible violation of TOS

### ISSUE THREAD

**Issue by cwilligv** [m-wrzr/populartimes#90]
Hello,

I just wanted to verify with you if the scraping part of the code would violate google's Terms of Service?

**10.1 (a) of ToS: "No access to APIs or Content except through the Service. You will not access the Maps API(s) or the Content except through the Service. For example, you must not access map tiles or imagery through interfaces or channels (including undocumented Google interfaces) other than the Maps API(s)."**

thanks

**Comment by tecfu** [comment]
@cwilligv Perhaps consult an attorney for legal questions about this software, rather than making a thinly veiled attempt to bait the author into justifying its legality.

**Comment by cwilligv** [comment]
@tecfu I'm not attempting to do anything here to the author. That's your interpretation mate. His work is of value and I was asking a valid question that was actually raised in stackoverflow (https://stackoverflow.com/questions/40456547/is-it-possible-to-get-the-popular-times-information-through-the-google-places)

Sorry about the misunderstanding, my intention wasn't at all to do harm to the author. I understand if he doesn't want to reply though.

cheers.

**Comment by alpalpalp** [comment]
I'm curious about this as well, i find the author's work very useful

**Comment by m-wrzr** [comment]
I'd suggest to star this issue (https://issuetracker.google.com/issues/35827350) and hope Google will add an offically supported version to get popular times data.

Sadly there is no ETA for it, no information during the last 5 years and with the decisions that the Places API is taking (e.g. removed Radar Search, increased API call cost) I don't see official support in the near future.

This wasn't added to PIP and is under little development because of ToS issues, but is somewhat tolerated i guess; if Google wants to stop it they can at any time. So don't use any of this data for commercial use, get some interesting data to do cool stuff with and push for official support. I will cite this from the README real quick

> The goal of this library is to provide an option to use Google Maps popular times data, until it is available via Google's API.

**Comment by MrUpsidown** [comment]
This is violating Google Maps TOS. Period. Because there is a feature request to provide this information via their APIs or because you say your library is a "temporary" solution until they provide it doesn't make it more (or less) acceptable.

And the fact that there is a feature request doesn't mean in any way that Google will or wants or will want to provide this data at any point in the future. There are feature requests in their tracker that are over 12 years old...

**Comment by federicotallis** [comment]
I just ran this tool and received an automated message from Google that they are suspending my account for 24 hours because of violating their terms of service specifically Section 3.2.3 on data scraping which reads as follows:

_(a)  No Scraping. Customer will not export, extract, or otherwise scrape Google Maps Content for use outside the Services. For example, Customer will not: (i) pre-fetch, index, store, reshare, or rehost Google Maps Content outside the services; (ii) bulk download Google Maps tiles, Street View images, geocodes, directions, distance matrix results, roads information, places information, elevation values, and time zone details; (iii) copy and save business names, addresses, or user reviews; or (iv) use Google Maps Content with text-to-speech services.__

**Comment by federicotallis** [comment]
Appealed my case to Google and they responded: 

_We can understand the interest in using Google Maps data to do research. Unfortunately, Google does not have the data rights to allow this use case as we do not own all the images that our API's serve. Additionally, creating new insights from imagery or data within the Google Maps API is creating a derivative work out of the data, which even for research purposes is against the Google Maps Terms of Service as outlined in Sections 3.2.2 and 3.2.3._

**Comment by cwilligv** [comment]
Thanks @federicotallis for sharing your outcome. It clarifies with evidence what I thought. This should be feedback for everyone else so they can plan accordingly. All the best in your research.
Cheers.

**Comment by StephenGilboy** [comment]
Since scraping websites is legal in the United States I would think a solution that does not use the API but instead grabs the data off of the maps html that is rendered would work and get around their ToS. You don't have to log in to Google maps to view the data so its publicly available. As for using it for commercial use, that might be tricky.

**Comment by pauldiez** [comment]
"but instead grabs the data off of the maps html that is rendered" - Isn't
that considered scraping though?

**Comment by StephenGilboy** [comment]
> "but instead grabs the data off of the maps html that is rendered" - Isn't that considered scraping though?

It is scraping and so at the moment, in the United States, it is legal. The issue with this library is that it's using Google's API to access data, which means Google is giving you authorization and so you have to use it within the terms they outline. But if you create a library that just loads up a google maps link, looks for the HTML section that has the popular times and scrape it that *should* be legal. 

> [The 9th Circuit rules in favor of HiQ, holding that scraping a public website likely does not violate the CFAA, even after website owner prohibits with a cease-and-desist letter; language strongly suggests CFAA only applies to bypassing authentication.](https://techstartups.com/2019/09/13/web-scraping-legal-us-court-says-scraping-public-data-website-without-permission-not-illegal/)

What you do with the data after you scrape it is where you will open another can of worms. Since the "popular times" isn't added by the owner of the business and comes from a proprietary algorithm it *could* be considered copyrighted.

**Comment by mick-net** [comment]
> > [The 9th Circuit rules in favor of HiQ, holding that scraping a public website likely does not violate the CFAA, even after website owner prohibits with a cease-and-desist letter; language strongly suggests CFAA only applies to bypassing authentication.](https://techstartups.com/2019/09/13/web-scraping-legal-us-court-says-scraping-public-data-website-without-permission-not-illegal/)
> 
> What you do with the data after you scrape it is where you will open another can of worms. Since the "popular times" isn't added by the owner of the business and comes from a proprietary algorithm it _could_ be considered copyrighted.

Good point, regarding the proprietary algorithm, but can't we consider it as publicly generated data? The popular data is generated by the public (not by Google) using tons of personal smartphones (who gave permission to Google to analyse it and publicly share it as 'public' data. Any thoughts? @StephenGilboy

**Comment by ejenk0** [comment]
The relevant issue has been marked "Accepted" on Google's issue tracker  as of yesterday: https://issuetracker.google.com/issues/35827350
